orc8r/cloud/go/obsidian/tests/unit_test_utils.go

```diff
@@ -122,6 +134,18 @@ func (j *jsonMarshaler) MarshalBinary() (data []byte, err error) {
 	return json.Marshal(j.v)
 }
 
+func YAMLMarshaler(v string) encoding.BinaryMarshaler {
```

Comment:
What you have right now is actually StringMarshaler

The first-best way to handle this is probably to have specs implement the encoding.BinaryMarshaler interface directly. Then you won't need this new test-oriented YAMLMarshaler

That is, add a Spec.MarshalBinary method. It can either replace or forward to the existing marshalToYAML private method. Then, you can just pass a spec directly to ExpectedResult, rather than having to first marshal it to string

Reply:
Ahhh got it, that's definitely a cleaner solution.